x/ibc/04-channel/keeper/handshake.go

```diff
@@ -87,64 +82,75 @@ func (k Keeper) ChanOpenTry(
 	proofInit commitment.ProofI,
 	proofHeight uint64,
 ) error {
-	_, found := k.GetChannel(ctx, portID, channelID)
+	// channel identifier and connection hop length checked on msg.ValidateBasic()
+
+	previousChannel, found := k.GetChannel(ctx, portID, channelID)
 	if found {
 		return sdkerrors.Wrap(types.ErrChannelExists, channelID)
+	} else if !(previousChannel.State == exported.INIT &&
```

Comment:
Can we separate this out into a helper function (perhaps in the spec as well)?

Reply:
It'd be helpful to have it on the spec 👍